src/main/java/baritone/utils/schematic/format/defaults/StructureNBT.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.utils.schematic.format.defaults;
+
+import baritone.utils.schematic.StaticSchematic;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.gen.structure.template.Template;
+
+import java.lang.reflect.Field;
+import java.util.List;
+
+/**
+ * @author Sam
+ * @since 03/11/2022
+ */
+public final class StructureNBT extends StaticSchematic {
+
+    public StructureNBT(NBTTagCompound schematic) {
+        Template template = new Template();
+        template.read(schematic);
+
+        BlockPos size = template.getSize();
+        this.x = size.getX();
+        this.y = size.getY();
+        this.z = size.getZ();
+
+        List<Template.BlockInfo> blocks;
+        try {
+            Field f = template.getClass().getDeclaredField("field_186270_a"); // Template.blocks
+            f.setAccessible(true);
+            blocks = (List<Template.BlockInfo>) f.get(template);
+        } catch (final Exception e) {
+            // This should never happen since SRG names are canonical
+            throw new IllegalStateException("reflection error");
+        }
```

Comment:
This is a no go. It only works when loading Baritone as a Forge mod in a non-dev enviroment.
Cases where Baritone has to work:
* Baritone dev env, mcp or official names
* Vanilla minecraft, obfuscated names
* Forge, srg names
* Fabric, yarn names
* As a library in other dev envs, mcp or official or yarn names

Of course it is an option to look up all possible names and check them all but that would mean manually validating the official, yarn, srg and obf name for every minecraft version to come. Instead you should use mixin so the mixin preprocessor and the transformers in Forge and fabric take care of the different mappings just like for any other name.

Look at the interfaces in `src/main/java/baritone/utils/accessor` and their implementations `src/launch/java/baritone/launch/mixins` for working and used examples. (and feel free to ask questions)

Reply:
Ooooh, very interesting, i'll look into it.
Kinda thought that hardcoding fieldnames was whack but didn't know about Mixins.

Reply:
I made a new commit but im not sure if i've done it right. It seems to work, but it feels like it shouldn't...

Reply:
It's just injecting a bit of code at runtime.
When the game is running mixin merges the `MixinTemplate` class into the `Template` class so it implements the interfaces and has the methods (`@Accessor` is a shorthand for a getter on the field with the corresponding name) and fields.

Reply:
I understand whats happening, but it just seems kinda weird since im never actually referencing the field that i want the getter to return, it just reads the function name and parses out the field name, which to me seems weird.

But hey, on the fabric wiki it said thats the way to do it, and it works, so who am i to judge ._.